### PR TITLE
feat: track average entry price in position engine

### LINKED ISSUE #10004: [Feature]: Add average entry price to position engine
# Feature Overview

**In order to** support the requirements for margin isolation 
**We will** add average entry price to the position engine
**So that** so we can calculate the amount of funds that need to be moved to the margin account when switching between modes.

### PR DESCRIPTION
**Pull request by ze97286**
Closes #10004

### CODE REVIEW

File: core/positions/engine.go [hunk 1]
```diff
@@ -612,3 +617,17 @@ func (e *Engine) remove(p *MarketPosition) {
...
+	if volume+addVolume == 0 || addPrice == nil {
+		return num.DecimalZero()
+	}
+
+	volumeDec := num.DecimalFromInt64(volume)
+	addVolumeDec := num.DecimalFromInt64(addVolume)
+	addPriceDec := num.DecimalFromUint(addPrice)
+
+	//	return ((vwap * float64(volume)) + (float64(addPrice) * float64(addVolume))) / (float64(volume) + float64(addVolume))
+	return vwap.Mul(volumeDec).Add(addPriceDec.Mul(addVolumeDec)).Div(volumeDec.Add(addVolumeDec))
```
Comment (jeremyletang) [inline core/positions/engine.go:632]: Any chance we could use Uints here instead?

Comment (ze97286) [reply core/positions/engine.go:632]: maybe lets see how it goes first performance wise and decide later?

Comment (jeremyletang) [reply core/positions/engine.go:632]: Alright.

Comment (ze97286) [reply core/positions/engine.go:632]: no, not alright. I changed my mind again.

Comment (jeremyletang) [reply core/positions/engine.go:632]: alright then.

Comment (ze97286) [reply core/positions/engine.go:632]: not so much because of data type, more because all of this is wrong

**Review by candida-d — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**